planner/core/fragment.go

```diff
@@ -115,20 +134,99 @@ func (f *Fragment) init(p PhysicalPlan) error {
 	return nil
 }
 
-func newFragment(s *PhysicalExchangeSender) (*Fragment, error) {
-	f := &Fragment{ExchangeSender: s}
-	s.Fragment = f
-	err := f.init(s)
-	return f, errors.Trace(err)
+func flattenPlanForUnionAll(stack []PhysicalPlan, forest *[]*PhysicalExchangeSender) error {
+	cur := stack[len(stack)-1]
+	switch x := cur.(type) {
+	case *PhysicalTableScan, *PhysicalExchangeReceiver: // This should be the leave node.
+		p, err := stack[0].Clone()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		*forest = append(*forest, p.(*PhysicalExchangeSender))
+		for i := 1; i < len(stack); i++ {
+			if _, ok := stack[i].(*PhysicalUnionAll); ok {
```

Comment:
Do we need to adjust the columns below and above this removed UnionAll to keep same UniqueID? or replace this UnionAll with a Projection?

Reply:
I think it's unnecessary, because TiFlash won't use `UniqueID`.